java/libs/src/main/java/org/physical_web/collection/PwsClient.java

```diff
@@ -33,36 +33,78 @@
  */
 public class PwsClient {
   private static final String DEFAULT_PWS_ENDPOINT = "https://url-caster.appspot.com";
-  private static final String RESOLVE_SCAN_PATH = "resolve-scan";
+  private static final int DEFAULT_PWS_VERSION = 1;
+  private static final String V1_RESOLVE_SCAN_PATH = "resolve-scan";
+  private static final String V2_RESOLVE_SCAN_PATH = "v1alpha1/urls:resolve";
+  private static final String UKNOWN_API_ERROR_MESSAGE = "Unknown API Version";
   private String mPwsEndpoint;
+  private String apiKey;
+  private int apiVersion;
   private List<Thread> mThreads;
 
   /**
    * Construct a PwsClient.
    */
   public PwsClient() {
-    this(DEFAULT_PWS_ENDPOINT);
+    this(DEFAULT_PWS_ENDPOINT, DEFAULT_PWS_VERSION);
   }
 
   /**
    * Construct a PwsClient.
    * @param pwsEndpoint The URL to send requests to.
+   * @param pwsApiVersion The API version the endpoint uses.
    */
-  public PwsClient(String pwsEndpoint) {
-    setPwsEndpoint(pwsEndpoint);
+  public PwsClient(String pwsEndpoint, int pwsApiVersion) {
+    this(pwsEndpoint, pwsApiVersion, "");
+  }
+
+  /**
+   * Construct a PwsClient.
+   * @param pwsEndpoint The URL to send requests to.
+   * @param pwsApiVersion The API version the endpoint uses.
+   * @param pwsApiKey The api key to access the endpoint.
+   */
+  public PwsClient(String pwsEndpoint, int pwsApiVersion, String pwsApiKey) {
+    setPwsEndpoint(pwsEndpoint, pwsApiVersion, pwsApiKey);
     mThreads = new ArrayList<>();
   }
 
   /**
-   * Set the URL for making PWS requests.
+   * Set the URL, the API version, the API Key for making PWS requests.
    * @param pwsEndpoint The new PWS endpoint.
+   * @param pwsApiVersion The new PWS API version.
+   * @param pwsApiKey The new PWS API key.
    */
-  public void setPwsEndpoint(String pwsEndpoint) {
+  public void setPwsEndpoint(String pwsEndpoint, int pwsApiVersion, String pwsApiKey) {
     mPwsEndpoint = pwsEndpoint;
+    apiVersion = pwsApiVersion;
+    apiKey = pwsApiKey;
```

Comment:
test if apiKey is null or empty and version is >=2. If so, throw a RuntimeException